### PR TITLE
Upgrade to netcoreapp22

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Comment by pranavkm** [comment]
Need to rebase on dev

File: Directory.Build.targets [hunk 1]
```diff
@@ -1,6 +1,8 @@
-﻿<Project>
+<Project>
   <PropertyGroup>
     <RuntimeFrameworkVersion Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">$(MicrosoftNETCoreApp20PackageVersion)</RuntimeFrameworkVersion>
```
Comment (pranavkm) [inline Directory.Build.targets]: Is this still needed

File: benchmarks/Microsoft.AspNetCore.Mvc.Performance/Microsoft.AspNetCore.Mvc.Performance.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+    <TargetFrameworks>net461</TargetFrameworks>
```
Comment (pranavkm) [inline benchmarks/Microsoft.AspNetCore.Mvc.Performance/Microsoft.AspNetCore.Mvc.Performance.csproj]: Incorrect

File: build/dependencies.props [hunk 3]
```diff
@@ -68,6 +68,7 @@
     <MicrosoftExtensionsWebEncodersPackageVersion>2.1.0-preview1-27644</MicrosoftExtensionsWebEncodersPackageVersion>
     <MicrosoftNETCoreApp20PackageVersion>2.0.0</MicrosoftNETCoreApp20PackageVersion>
     <MicrosoftNETCoreApp21PackageVersion>2.1.0-preview1-25907-02</MicrosoftNETCoreApp21PackageVersion>
```
Comment (pranavkm) [inline build/dependencies.props]: Remove?

File: build/repo.props [hunk 4]
```diff
@@ -14,6 +14,6 @@
 
   <ItemGroup>
     <DotNetCoreRuntime Include="$(MicrosoftNETCoreApp20PackageVersion)" />
```
Comment (pranavkm) [inline build/repo.props]: Remove?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj [hunk 5]
```diff
@@ -1,12 +1,12 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
   <Import Project="..\..\src\Microsoft.AspNetCore.Mvc.Testing\build\netstandard2.0\Microsoft.AspNetCore.Mvc.Testing.targets" />
 
   <PropertyGroup>
     <TargetFrameworks>$(StandardTestTfms)</TargetFrameworks>
     <!-- Removing this is blocked by https://github.com/aspnet/Mvc/issues/5822 -->
-    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp2.1' OR '$(TargetFramework)' == 'netcoreapp2.0'  ">$(PackageTargetFallback);portable-net451+win8</PackageTargetFallback>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp2.2' OR '$(TargetFramework)' == 'netcoreapp2.0'  ">$(PackageTargetFallback);portable-net451+win8</PackageTargetFallback>
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: This looks like a super old branch - https://github.com/aspnet/Mvc/commit/bab520f4dd358dfd6fa22c727670a7a06a8b25f9#diff-e8c2febb82a99652effd92b853ac509dL5

File: build/repo.props (hunk 4 above)
Comment (ryanbrandenburg) [reply build/repo.props]: I talked it over with @natemcmaster and we decided that due to the complexities of figuring out where to keep and where to delete these things across 50 repos, we're just going to leave them for now. They don't hurt anything, and I've filed aspnet/Universe#1153 to remove them in the future.

**Comment by pranavkm** [comment]
@dotnet-bot test Ubuntu 16.04 Release Build please

**Pull request merged**